File: Juego/Invasiones/fuente/Programa.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Invasiones
{
    /// <summary>
    /// La clase que contiene el punto de entrada de la aplicación.
    /// </summary>
    public class Programa
    {
        /// <summary>
        /// La cantidad de FPS por default.
        /// </summary>
        public const int FPS_POR_DEFECTO = 20;

        /// <summary>
        /// El path donde esta a la data.
        /// </summary>
        public const string PATH_DATA_DEL_PROG = "..\\..\\..\\Data";

        /// <summary>
        /// El path donde esta a la data.
        /// </summary>
        public const string PATH_ICONO = "imagenes\\icono.png";

        /// <summary>
        /// El path donde esta la data.
        /// </summary>
        public const string PATH_DATA = "Data";

        public const string PATH_ESCENARIOS = "escenarios";

		public const string PATH_NIVEL = "nivel";

        public const string ARCHIVO_XML_TEXTOS = "strings.xml";

        public const string ARCHIVO_XML_RECURSOS = "res.xml";

        /// <summary>
        /// El ancho de la pantalla.
        /// </summary>
        public const short ANCHO_DE_LA_PANTALLA = 1024;

        /// <summary>
        /// El alto de la pantalla.
        /// </summary>
        public const short ALTO_DE_LA_PANTALLA = 768;

        /// <summary>
        /// Si el juego es fullcreen o no.. 
        /// </summary>
#if (DEBUG)
        public const bool FULLSCREEN = false;
#else
		public const bool FULLSCREEN = true;
#endif


		static void Main(string[] args)
        {
            GameFrame gameFrame = new GameFrame(ANCHO_DE_LA_PANTALLA, ALTO_DE_LA_PANTALLA, FPS_POR_DEFECTO, FULLSCREEN);
        }

    }
}

File: macOS/Invasiones/Invasiones/Programa.swift
//
//  Programa.swift
//  Invasiones
//
//  Created by Lucia Medina Fretes on 06.04.26.
//
//  Port of Programa.cs — global application constants.
//  Entry point is AppDelegate/GameScene on macOS; only constants here.
//

import Foundation

enum Programa {

    static let FPS_POR_DEFECTO: Int = 20

    static let PATH_DATA       = "data"
    static let PATH_ESCENARIOS = "escenarios"
    static let PATH_NIVEL      = "nivel"
    static let PATH_ICONO      = "imagenes/icono.png"

    static let ARCHIVO_XML_TEXTOS   = "strings.xml"
    static let ARCHIVO_XML_RECURSOS = "res.xml"

    static let ANCHO_DE_LA_PANTALLA: Int = 1024
    static let ALTO_DE_LA_PANTALLA:  Int = 768

#if DEBUG
    static let FULLSCREEN = false
#else
    static let FULLSCREEN = true
#endif
}
